x-pack/plugins/alerting/server/usage/lib/get_telemetry_from_event_log.ts

```diff
@@ -0,0 +1,620 @@
+        },
+      },
+    };
+
+    logger.debug(`query for getExecutionTimeoutsPerDayCount - ${JSON.stringify(query)}`);
+    const results = await esClient.search(query);
+
+    logger.debug(`results for getExecutionTimeoutsPerDayCount query - ${JSON.stringify(results)}`);
+
+    const aggregations = results.aggregations as {
+      by_rule_type_id: AggregationsTermsAggregateBase<AggregationsStringTermsBucketKeys>;
+    };
+
+    const totalTimedoutExecutionsCount =
+      typeof results.hits.total === 'number' ? results.hits.total : results.hits.total?.value;
+
+    return {
+      countExecutionTimeouts: totalTimedoutExecutionsCount ?? 0,
+      countExecutionTimeoutsByType: parseSimpleRuleTypeBucket(aggregations.by_rule_type_id.buckets),
+    };
+  } catch (err) {
+    logger.warn(
+      `Error executing alerting telemetry task: getExecutionsTimeoutsPerDayCount - ${JSON.stringify(
+        err
+      )}`
+    );
+    return {
+      countExecutionTimeouts: 0,
+      countExecutionTimeoutsByType: {},
+    };
+  }
+}
+
+/**
+ * Bucket format:
+ * {
+ *   key: '.index-threshold',             // rule type id
+ *   doc_count: 78,                       // count of number of executions
+ *   avg_es_search_duration: {            // average es search duration across executions
+ *     value: 40.76056338028169,
+ *   },
+ *   percentile_alerts: {                 // stats for number of alerts created across executions
+ *     values: {
+ *       '50.0': 1,
+ *       '95.0': 1,
+ *       '99.0': 1,
+ *     },
+ *   },
+ *   execution_failures: {
+ *     doc_count: 7,                      // count of number of failed executions
+ *     by_reason: {
+ *       doc_count_error_upper_bound: 0,
+ *       sum_other_doc_count: 0,
+ *       buckets: [
+ *         {
+ *           key: 'execute',              // breakdown of reason for execution failures
+ *           doc_count: 4,
+ *         },
+ *         {
+ *           key: 'decrypt',
+ *           doc_count: 3,
+ *         },
+ *       ],
+ *     },
+ *   },
+ *   percentile_scheduled_actions: {      // stats for number of actions generated across executions
+ *     values: {
+ *       '50.0': 0,
+ *       '95.0': 0,
+ *       '99.0': 0,
+ *     },
+ *   },
+ *   avg_execution_time: {                // average execution time in nanoseconds across executions
+ *     value: 100576923.07692307,
+ *   },
+ *   avg_total_search_duration: {         // average total search duration across executions
+ *     value: 43.74647887323944,
+ *   },
+ * }
+ */
+
+export function parseRuleTypeBucket(
+  buckets: GetExecutionCountsAggregationBucket[]
+): Pick<
+  GetExecutionsPerDayCountResults,
+  | 'countRuleExecutionsByType'
+  | 'avgExecutionTimeByType'
+  | 'avgEsSearchDurationByType'
+  | 'avgTotalSearchDurationByType'
+  | 'generatedActionsPercentilesByType'
+  | 'alertsPercentilesByType'
+> {
+  return (buckets ?? []).reduce(
+    (summary, bucket) => {
+      const ruleType: string = replaceDotSymbols(bucket.key);
```

Comment:
do we need `bucket?` on this line and one below, like we do for the others below that?

Reply:
Updated here: https://github.com/elastic/kibana/pull/134769/commits/a327793511e0b02c036282126b47782b6031041b